src/stream/mod.rs

```diff
@@ -411,45 +289,33 @@ impl ::std::fmt::Display for FilterLevel {
 /// Represents a `TwitterStream` before it is started.
 pub struct StreamBuilder {
     url: &'static str,
-    with_follows: Option<bool>,
-    all_replies: bool,
+    follow: BTreeSet<u64>,
+    track: BTreeSet<String>,
     filter_level: Option<FilterLevel>,
 }
 
 impl StreamBuilder {
-    fn new(url: &'static str) -> StreamBuilder {
+    fn new(url: &'static str) -> Self {
         StreamBuilder {
             url: url,
-            with_follows: None,
-            all_replies: false,
+            follow: BTreeSet::new(),
+            track: BTreeSet::new(),
             filter_level: None,
         }
     }
 
-    /// For User Streams, sets whether to include posts from just the authenticated user or from
-    /// the accounts they follow as well.
-    ///
-    /// By default, this is set to `true`, meaning the stream will include posts from the accounts
-    /// the user follows. This makes the stream act like the user's timeline.
-    pub fn with_follows(self, with_follows: bool) -> StreamBuilder {
-        StreamBuilder {
-            with_follows: Some(with_follows),
-            ..self
-        }
+    /// List of user IDs indicating the users whose Tweets should be delivered on the stream
+    pub fn follow(mut self, to_follow: &[u64]) -> Self {
+        self.follow.extend(to_follow.iter());
+        self
     }
 
-    /// For User Streams, sets whether to return all @replies by followed users, or just those
-    /// which are also to accounts the authenticated user follows.
-    ///
-    /// By default, user streams will only emit @replies if the authenticated user follows both the
-    /// account that posted it *and the account they are replying to*. This mirrors the user's home
-    /// timeline. By setting this to `true`, you can see all posts by the followed accounts,
-    /// regardless of whether they're replying to someone the authenticated user is not following.
-    pub fn all_replies(self, all_replies: bool) -> StreamBuilder {
-        StreamBuilder {
-            all_replies: all_replies,
-            ..self
-        }
+    /// List of phrases which will be used to determine what Tweets will be delivered on the stream.
+    /// A phrase may be one or more terms separated by spaces, and a phrase will match if all
+    /// of the terms in the phrase are present in the Tweet, regardless of order and ignoring case.
+    pub fn track(mut self, to_track: &[&str]) -> Self {
```

Comment:
The type of `to_track` might need some tweaking, since it's difficult to turn a `Vec<String>` into a `&[&str]`. This is why i used `IntoIterator` on the `lookup` functions.

Using something like this will allow you to take `&[&str]` and `&[String]`:

```rust
pub fn track<I: IntoIterator<Item=S>, S: AsRef<str>>(mut self, to_track: I) -> Self {
    self.track.extend(to_track.into_iter().map(|s| s.as_ref().to_string()));
    self
}
```

Reply:
Fixed